src/app/designationEditor/designationEditor.component.js

```diff
@@ -0,0 +1,220 @@
+import sessionService, {Roles} from 'common/services/session/session.service';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import designationEditorService from 'common/services/api/designationEditor.service';
+
+import titleModalController from './titleModal/title.modal';
+import pageOptionsModalController from './pageOptionsModal/pageOptions.modal';
+import photoModalController from './photoModal/photo.modal';
+import textEditorModalController from './textEditorModal/textEditor.modal';
+import websiteModalController from './websiteModal/website.modal';
+
+import template from './designationEditor.tpl';
+import titleModalTemplate from './titleModal/titleModal.tpl';
+import pageOptionsModalTemplate from './pageOptionsModal/pageOptionsModal.tpl';
+import photoModalTemplate from './photoModal/photoModal.tpl';
+import textEditorModalTemplate from './textEditorModal/textEditorModal.tpl';
+import websiteModalTemplate from './websiteModal/websiteModal.tpl';
+
+let componentName = 'designationEditor';
+
+class DesignationEditorController {
+
+  /* @ngInject */
+  constructor( $q, $uibModal, $location, $window, envService, sessionService, sessionEnforcerService, designationEditorService ) {
+    this.sessionService = sessionService;
+    this.sessionEnforcerService = sessionEnforcerService;
+    this.designationEditorService = designationEditorService;
+
+    this.imgDomain = envService.read('imgDomain');
+    this.imgDomainDesignation = envService.read('imgDomainDesignation');
+
+    this.$location = $location;
+    this.$q = $q;
+    this.$uibModal = $uibModal;
+    this.$window = $window;
+  }
+
+  $onInit() {
+    this.designationNumber = this.$location.search().d;
+
+    this.enforcerId = this.sessionEnforcerService( [Roles.registered], {
+      [EnforcerCallbacks.signIn]: () => {
+        this.getDesignationContent();
+      },
+      [EnforcerCallbacks.cancel]: () => {
+        // Authentication failure
+        this.$window.location = '/';
+      }
+    }, EnforcerModes.session );
+  }
+
+  $onDestroy() {
+    // Destroy enforcer
+    this.sessionEnforcerService.cancel( this.enforcerId );
+  }
+
+  getDesignationContent() {
+    if(!this.designationNumber){ return; }
+    this.loadingOverlay = true;
+
+    return this.$q.all([
+      //get designation content
+      this.designationEditorService.getContent(this.designationNumber).then((response) => {
+        this.designationContent = response.data;
+      }),
+
+      //get designation photos
+      this.designationEditorService.getPhotos(this.designationNumber).then((response) => {
+        this.designationPhotos = response.data;
+      })
+    ]).then(() => {
+      this.loadingOverlay = false;
+    });
+  }
+
+  editTitle() {
+    let modalOptions = {
+      templateUrl: titleModalTemplate.name,
+      controller: titleModalController.name,
```

Comment:
I like the idea of using the component syntax for all of these.  I don't feel that you need to change it if you don't want to but I think in the future it would be nice to use components. You can look at [existingPaymentMethods.component.js#L67-L79](https://github.com/CruGlobal/give-web/blob/master/src/app/checkout/step-2/existingPaymentMethods/existingPaymentMethods.component.js#L67-L79) and [paymentMethodForm.modal.component.js#L32-L33](https://github.com/CruGlobal/give-web/blob/master/src/common/components/paymentMethods/paymentMethodForm/paymentMethodForm.modal.component.js#L32-L33) for an example.